amalgamation/python/mxnet_predict.py

```diff
@@ -160,10 +249,18 @@ def forward(self, **kwargs):
         >>> predictor.forward(data=mydata)
         >>> out = predictor.get_output(0)
         """
+        if self.type_dict and len(self.type_dict) != len(kwargs.items()):
+            raise ValueError("number of kwargs should be same as len of type_dict" \
+                             "Please check your forward pass inputs" \
+                             "or type_dict passed to Predictor instantiation")
+
         for k, v in kwargs.items():
             if not isinstance(v, np.ndarray):
                 raise ValueError("Expect numpy ndarray as input")
-            v = np.asarray(v, dtype=np.float32, order='C')
+            if self.type_dict and k in self.type_dict:
+                v = np.asarray(v, dtype=self.type_dict[k], order='C')
+            else:
+                v = np.asarray(v, dtype=np.float32, order='C')
```

Comment:
nit: Will be better to keep all the dtype, including default ie., np.float32 in the map you are maintaing and remove all explicit np.dtype ?

Reply:
not sure i understand. can you elaborate.

Reply:
I meant, wherever there are defaults we explicitly call 'np.float32'. In other places we use a common lookup with _DTYPE_MX_TO_NP. So I was suggesting if you want to consider moving np.float32 as a default in these (_DTYPE_MX_TO_NP) look up table so there are no hard coded np.float32.

Reply:
we use the lookup table to map the dtype int, which comes from mxnet backend to numpy dtype. For cases, we are not determining it from backend dtype I think it is better to use np.float32 rather than _DTYPE_MX_TO_NP[0].